clippy_utils/src/macros.rs

```diff
@@ -425,6 +425,38 @@ pub fn find_format_arg_expr<'hir, 'ast>(
     .ok_or(&target.expr)
 }
 
+/// Finds and extracts all format arguments from a format-like macro call.
+///
+/// `callback` can also be passed, to process the [`FormatArgs`] manually
+/// if needed, without calling [`find_format_args`] separately.
+///
+/// ```ignore
+/// // vvvvv any format-like macro
+/// println!("Hello, {}!", "ferris")
+/// //                     ^^^^^^^^ returns these expressions
+/// ```
+pub fn collect_format_args<'hir>(
+    cx: &LateContext<'_>,
+    callback: impl FnOnce(&FormatArgs),
+    start: &'hir Expr<'hir>,
+    expn_id: ExpnId,
+) -> Vec<&'hir Expr<'hir>> {
```

Comment:
It'd want to be  `callback: impl FnOnce(&FormatArgs, Vec<...>)` rather than returning the `Vec`, lints generally want access to both at the same time

Reply:
Oh, sorry for misunderstanding then. Although I'm not quite sure how to apply it in the author lint, if we don't return the args, it wouldn't quite fit into the if-chain author lint tries to build, i. e.
```rs
   && macros::collect_from_args(cx, |args, _fmt_args| { if /* another if-chain */ { /* report your lint here */ } }, e, macro_call.expn)
```
\- this does not only introduce another level on indentation, but also requires quite a bit of code refactoring to support, which is, in my opinion, quite a bit out of scope for the author lint.

If you have any ideas on how to mitigate this, I would be happy to hear, otherwise I can close the PR and submit an issue on this.

Reply:
For `author` hmm that's tricky, I guess you could always print a comment saying to take a look at the function